libraries/libduck/Path.cpp

```diff
@@ -78,12 +80,19 @@ ResultRet<std::vector<DirectoryEntry>> Path::get_directory_entries() const {
 	std::vector<DirectoryEntry> entries;
 
 	struct dirent* entry;
-	while((entry = readdir(dir)) != NULL)
+	while((entry = readdir(dir)) != NULL) {
 		if(std::string(entry->d_name) != "." && std::string(entry->d_name) != "..")
 			entries.emplace_back(*this, entry);
-
+	}
 	closedir(dir);
 
+	std::sort(entries.begin(), entries.end(), [](auto const& lhs, auto const& rhs) {
+		return lhs.name() < rhs.name();
+	});
+	std::sort(entries.begin(), entries.end(), [](auto const& lhs, auto const& rhs) {
+		return lhs.type() > rhs.type();
+	});
+
```

Comment:
I don't know if we should be sorting directory entries by default - this could incur unnecessary performance costs when iterating over directories where sorting isn't important. I think we should probably extract this logic to `ls.cpp` for now.

What we _could_ do is override `operator<` for `DirectoryEntry` so that it's easier to just call `std::sort` on it without writing a custom comparison function.